pulp_container/app/registry_api.py

```diff
@@ -392,10 +392,24 @@ def get_dr_push(self, request, path, create=False):
 
     def create_dr(self, path, request):
         with transaction.atomic():
-            repository = serializers.ContainerPushRepositorySerializer.get_or_create({"name": path})
-            distribution = serializers.ContainerDistributionSerializer.get_or_create(
-                {"base_path": path, "name": path}, {"repository": get_url(repository)}
-            )
+            try:
+                repository = serializers.ContainerPushRepositorySerializer.get_or_create(
+                    {"name": path}
+                )
+            except models.ContainerPushRepository.DoesNotExist:
+                raise RepositoryInvalid(name=path, message="Repository is read-only.")
+
+            try:
+                distribution = serializers.ContainerDistributionSerializer.get_or_create(
+                    {"base_path": path, "name": path}, {"repository": get_url(repository)}
+                )
+            except models.ContainerDistribution.DoesNotExist:
+                raise RepositoryInvalid(
+                    name=path,
+                    message="Cannot push to "
+                    + path
+                    + ". Another plugin with the same distribution name already exists.",
```

Comment:
What if there is a pulp-container pull-through distribution that has the same name? Is it still another plugin? 😈

Reply:
Good catch!